elb_log_ingestor/elb_log_parse.py

```diff
@@ -0,0 +1,330 @@
+        \ (?P<time>[^ ]*)  # leading backslash escapes the leading space
+        \ (?P<elb>[^ ]*)
+        \ (?P<client_ip>[^ ]*):(?P<client_port>[0-9]*)
+        \ (?P<target_ip>[^ ]*)[:-](?P<target_port>[0-9]*)
+        \ (?P<request_processing_time>[-.0-9]*)
+        \ (?P<target_processing_time>[-.0-9]*)
+        \ (?P<response_processing_time>[-.0-9]*)
+        \ (?P<elb_status_code>[-0-9]*)
+        \ (?P<target_status_code>[-0-9]*)
+        \ (?P<received_bytes>[-0-9]*)
+        \ (?P<sent_bytes>[-0-9]*)
+        \ "(?P<request_verb>[^ ]*)
+        \ (?P<request_url>[^ ]*)
+        \ (?P<request_proto>-|[^ ]*)\ ?"
+        \ "(?P<user_agent>[^"]*)"
+        \ (?P<ssl_cipher>[A-Z0-9-]+)
+        \ (?P<ssl_protocol>[A-Za-z0-9.-]*)
+        \ (?P<target_group_arn>[^ ]*)
+        \ "(?P<trace_id>[^"]*)"
+        \ "(?P<domain_name>[^"]*)"
+        \ "(?P<chosen_cert_arn>[^"]*)"
+        \ (?P<matched_rule_priority>[-.0-9]*)
+        \ (?P<request_creation_time>[^ ]*)
+        \ "(?P<actions_executed>[^"]*)"
+        \ "(?P<redirect_url>[^"]*)
+        "(?P<lambda_error_reason>$|\ "[^ ]*")  # probably never used
+        (?P<new_field>.*)  # probably never used
+        """,
+    re.VERBOSE,
+)
+
+ELB_LOG_LINE_REGEX = re.compile(
+    r"""
+        (?P<time>[^ ]*)
+        \ (?P<elb>[^ ]*)
+        \ (?P<client_ip>[^ ]*):(?P<client_port>[0-9]*)
+        \ (?P<target_ip>[^ ]*)[:-](?P<target_port>[0-9]*)
+        \ (?P<request_processing_time>[-.0-9]*)
+        \ (?P<target_processing_time>[-.0-9]*)
+        \ (?P<response_processing_time>[-.0-9]*)
+        \ (?P<elb_status_code>|[-0-9]*)
+        \ (?P<target_status_code>-|[-0-9]*)
+        \ (?P<received_bytes>[-0-9]*)
+        \ (?P<sent_bytes>[-0-9]*)
+        \ "(?P<request_verb>[^ ]*)
+        \ (?P<request_url>[^ ]*)
+        \ (?P<request_proto>-|[^ ]*)\ ?"
+        \ "(?P<user_agent>[^"]*)"
+        \ (?P<ssl_cipher>[A-Z0-9-]+)
+        \ (?P<ssl_protocol>[A-Za-z0-9.-]*)
+        """,
+    re.VERBOSE,
+)
+# map of key name to desired type constructor
+ALB_LOGS_FIELD_TYPES = {
+    "type": str,
+    "time": timestamp_to_timestamp,
+    "elb": str,
+    "client_ip": str,
+    "client_port": int,
+    "target_ip": str,
+    "target_port": int,
+    "request_processing_time": float,
+    "target_processing_time": float,
+    "response_processing_time": float,
+    "elb_status_code": int,
+    "target_status_code": int,
+    "received_bytes": int,
+    "sent_bytes": int,
+    "request_verb": str,
+    "request_url": str,
+    "request_proto": str,
+    "user_agent": str,
+    "ssl_cipher": str,
+    "ssl_protocol": str,
+    "target_group_arn": str,
+    "trace_id": str,
+    "domain_name": str,
+    "chosen_cert_arn": str,
+    "matched_rule_priority": str,
+    "request_creation_time": timestamp_to_timestamp,
+    "actions_executed": str,
+    "redirect_url": str,
+    "lambda_error_reason": str,
+    "new_field": str,
+}
+
+ALB = "alb"
+ELB = "elb"
+
+
+class LogParser:
```

Comment:
Why aren't all the methods in here not part of the class? They seem related.

Reply:
Reasonable. So long as things are well documented I'm not sure I care that much since Python is flexible like that.